File: FactorySimulation/src/http/HttpBroker.java
package http;

import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.URL;

import factory.Factory;
import factory.FactoryLine;
import factory.Machine;
import factory.Unit;

public class HttpBroker extends Thread {
	
	

	private String data;
	private String backendAddress ="http://localhost:8080/sendmachine/";
	private String PostBody;
	
	public static int counter;
	
	public void sendFactory(Factory factory, long timeStamp) {
		data ="";

		PostBody ="";
		for(FactoryLine fl  :factory.getFactoryLines()) {
			for(Unit u : fl.getUnits()) {
				for(Machine m : u.getMachines()) {
					addBody(m.getId(), m.getStatus(), m.getCurrentPower(), m.getWorkingOn(),timeStamp);
					
				}
			}
		}
		
		this.start();
		
		
		
	}
	
	
	private void addBody(int id, int s, float p , int w, long t) {
	
		if(data.equals("")) {
			String st="{\"machineId\": "+ id+", " +"\"stateCode\": " + s +", " + "\"power\": " +p +", " +"\"workingOn\": " +w+ ", " +"\"timestamp\": " +t +"}";
			
			data +=st;

		}else {
			data +=(" ,{"+ "\"machineId\": "+ id+", " +"\"stateCode\": " + s +", " + "\"power\": " +p +", " +"\"workingOn\": " +w+ ", " +"\"timestamp\": " +t+"}" ); 
		}
		
		
		
	}
	
	
	
	private void sendRequest() {
		
		 PostBody = "[ " +data +"]";
		
		
		if(counter %1 == 0) {
			System.out.println( PostBody);
			counter =0;
			
		}
			counter++;
			
		try {
		
			URL url = new URL(backendAddress);
			HttpURLConnection httpCon = (HttpURLConnection) url.openConnection();
			httpCon.setRequestProperty("Accept-Encoding", "application/json");
			httpCon.setRequestProperty("Content-Type", "application/json");
			 httpCon.setDoInput(true);
			httpCon.setDoOutput(true);
			httpCon.setRequestMethod("PUT");
			OutputStreamWriter out = new OutputStreamWriter(
			    httpCon.getOutputStream());
			out.write(PostBody);
			out.flush();
			
		
			httpCon.getInputStream();
			
		}catch(Exception e) {
			System.out.println("SENDING FAILED");
		}
	
		
		

		PostBody ="";
	}
	
	@Override	
	public void run() {
		
		sendRequest();
	
	
	}

}

File: FactorySimulation/C# Simulation/FactorySimulation/HttpBroker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace FactorySimulation
{
    class HttpBroker
    {

	

		private String _data;
		private String _backendAddress = "http://192.168.2.102:8080/sendmachine/";
		private  HttpClient client;

		public static int _counter;

		public void SendFactory(Factory factory, long timeStamp)
		{
			_data = "";

			
			foreach (Line fl  in factory.GetFactoryLines())
			{
				foreach (Unit u in fl.GetUnits())
				{
					foreach (Machine m in u.GetMachines())
					{
						AddBody(m.GetId(), m.GetStatus(), m.GetCurrentPower(), m.GetWorkingOn(), timeStamp);

					}
				}
			}



			// For each PUT a new Thread will be created
			Thread t = new Thread(SendRequest);
			t.Start();

		}


		private void AddBody(int id, int s, float p, int w, long t)
		{

			if (_data.Equals(""))
			{
				String st = "{\"machineId\": " + id + ", " + "\"stateCode\": " + s + ", " + "\"power\": " + p + ", " + "\"workingOn\": " + w + ", " + "\"timestamp\": " + t + "}";

				_data += st;

			}
			else
			{
				_data += (" ,{" + "\"machineId\": " + id + ", " + "\"stateCode\": " + s + ", " + "\"power\": " + p + ", " + "\"workingOn\": " + w + ", " + "\"timestamp\": " + t + "}");
			}



		}



		private async void  SendRequest()
		{
			String putBody = "["+ _data + "]";
			Console.WriteLine("SEND:"+  putBody);
			StringContent content = new StringContent(putBody, Encoding.UTF8,"application/json");

			
		

            try
			{
				using HttpResponseMessage message = await client.PutAsync(_backendAddress, content);
				message.EnsureSuccessStatusCode();

            }
            catch(Exception)
            {
				Console.Write("SENDING FAILED");
            }
			
		

		}

		
		public HttpBroker()
        {
			client = new HttpClient();
			
		}


	    

	}
}
